sidx.go

```diff
@@ -0,0 +1,39 @@
+package mp4
+
+import "fmt"
+
+func BoxTypeSidx() BoxType { return StrToBoxType("sidx") }
+
+func init() {
+	AddBoxDef(&Sidx{}, 0)
+}
+
+// Sidx is ISOBMFF Sidx box type
+type Sidx struct {
+	FullBox    `mp4:"extend"`
+	ReferenceID uint32 `mp4:"size=32"`
+	TimeScale uint32 `mp4:"size=32"`
+	EarliestPresentationTime uint32 `mp4:"size=32"`
+	FirstOffset uint32 `mp4:"size=32"`
+	Reserved uint16 `mp4:"size=16"`
+	ReferenceCount uint16 `mp4:"size=16"`
+	References []struct {
+		TypeSize uint32 `mp4:"size=32"`
+		SubSegmentDuration uint32 `mp4:"size=32"`
+		SapStartsTypeDeltaTime uint32 `mp4:"size=32"`
```

Comment:
Reference entity has defined in spec as follows:

```
      bit (1)           reference_type;
      unsigned int(31)  referenced_size;
      unsigned int(32)  subsegment_duration;
      bit(1)            starts_with_SAP;
      unsigned int(3)   SAP_type;
      unsigned int(28)  SAP_delta_time;
```

So, I think that this struct may be like following code.

```
		ReferencedType bool           `mp4:"size=1"`
		ReferencedSize uint32         `mp4:"size=31"`
                SubsegmentDuration uint32     `mp4:"size=32"`
		StartsWithSAP bool            `mp4:"size=1"`
		SAPType uint8                 `mp4:"size=3"`
		SapStartsTypeDeltaTime uint32 `mp4:"size=28"`
```

Do you refer to older spec?